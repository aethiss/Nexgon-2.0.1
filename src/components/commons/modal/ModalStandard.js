/* eslint-disable arrow-body-style */
import React from 'react'
import PropsTypes from 'prop-types'

// Style
import { Modal, Text, TouchableHighlight, View, StyleSheet } from 'react-native'

const style = StyleSheet.create({
  viewContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  viewContent: {
    backgroundColor: '#aacadd',
    width: '90%',
    height: '50%',
  },
})

const ModalStandard = ({ onClose, modalVisible, trasparent }) => {
  return (
    <Modal
      animationType="slide"
      transparent={trasparent}
      visible={modalVisible}
      onRequestClose={onClose}
    >
      <View style={style.viewContainer}>
        <View style={style.viewContent}>
          <Text>Hello World!</Text>
          <TouchableHighlight
            onPress={onClose}
          >
            <Text>Hide Modal</Text>
          </TouchableHighlight>
        </View>
      </View>
    </Modal>
  )
}

ModalStandard.propTypes = {
  onClose: PropsTypes.func.isRequired,
  modalVisible: PropsTypes.bool.isRequired,
  trasparent: PropsTypes.bool,
}

ModalStandard.defaultProps = {
  trasparent: false,
}

export default ModalStandard
