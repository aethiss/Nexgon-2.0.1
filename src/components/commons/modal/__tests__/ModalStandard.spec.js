/* eslint-disable no-undef */
import React from 'react'
import Adapter from 'enzyme-adapter-react-16'
import { shallow, configure } from 'enzyme'
import ModalStandard from '../ModalStandard'

configure({ adapter: new Adapter() })

describe('Register', () => {
  const onCloseModalMock = jest.fn()

  describe('renders', () => {
    it('default render', () => {
      const wrapper = shallow(<ModalStandard
        onClose={onCloseModalMock}
        modalVisible
      />)
      expect(wrapper.dive()).toMatchSnapshot()
    })
  })
  describe('actions', () => {
    it('should call close callback modal onPress ', () => {
      const component = shallow(<ModalStandard
        onClose={onCloseModalMock}
        modalVisible
      />)

      component.find('TouchableHighlight').simulate('press')
      expect(onCloseModalMock).toHaveBeenCalled()
    })
  })
})
