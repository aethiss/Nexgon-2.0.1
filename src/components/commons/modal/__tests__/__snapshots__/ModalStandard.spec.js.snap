// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register renders default render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Component
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[MockFunction]}
    transparent={false}
    visible={true}
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Component
        style={
          Object {
            "backgroundColor": "#aacadd",
            "height": "50%",
            "width": "90%",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Hello World!
        </Text>
        <TouchableHighlight
          activeOpacity={0.85}
          delayPressOut={100}
          onPress={[MockFunction]}
          underlayColor="black"
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            Hide Modal
          </Text>
        </TouchableHighlight>
      </Component>
    </Component>
  </Component>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "animationType": "slide",
      "children": <Component
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Component
          style={
            Object {
              "backgroundColor": "#aacadd",
              "height": "50%",
              "width": "90%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            Hello World!
          </Text>
          <TouchableHighlight
            activeOpacity={0.85}
            delayPressOut={100}
            onPress={[MockFunction]}
            underlayColor="black"
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              Hide Modal
            </Text>
          </TouchableHighlight>
        </Component>
      </Component>,
      "hardwareAccelerated": false,
      "onRequestClose": [MockFunction],
      "transparent": false,
      "visible": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Component
          style={
            Object {
              "backgroundColor": "#aacadd",
              "height": "50%",
              "width": "90%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            Hello World!
          </Text>
          <TouchableHighlight
            activeOpacity={0.85}
            delayPressOut={100}
            onPress={[MockFunction]}
            underlayColor="black"
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              Hide Modal
            </Text>
          </TouchableHighlight>
        </Component>,
        "style": Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              Hello World!
            </Text>,
            <TouchableHighlight
              activeOpacity={0.85}
              delayPressOut={100}
              onPress={[MockFunction]}
              underlayColor="black"
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                Hide Modal
              </Text>
            </TouchableHighlight>,
          ],
          "style": Object {
            "backgroundColor": "#aacadd",
            "height": "50%",
            "width": "90%",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Hello World!",
              "ellipsizeMode": "tail",
            },
            "ref": null,
            "rendered": "Hello World!",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeOpacity": 0.85,
              "children": <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                Hide Modal
              </Text>,
              "delayPressOut": 100,
              "onPress": [MockFunction],
              "underlayColor": "black",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Hide Modal",
                "ellipsizeMode": "tail",
              },
              "ref": null,
              "rendered": "Hide Modal",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "Modal",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "animationType": "slide",
        "children": <Component
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Component
            style={
              Object {
                "backgroundColor": "#aacadd",
                "height": "50%",
                "width": "90%",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              Hello World!
            </Text>
            <TouchableHighlight
              activeOpacity={0.85}
              delayPressOut={100}
              onPress={[MockFunction]}
              underlayColor="black"
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                Hide Modal
              </Text>
            </TouchableHighlight>
          </Component>
        </Component>,
        "hardwareAccelerated": false,
        "onRequestClose": [MockFunction],
        "transparent": false,
        "visible": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Component
            style={
              Object {
                "backgroundColor": "#aacadd",
                "height": "50%",
                "width": "90%",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              Hello World!
            </Text>
            <TouchableHighlight
              activeOpacity={0.85}
              delayPressOut={100}
              onPress={[MockFunction]}
              underlayColor="black"
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                Hide Modal
              </Text>
            </TouchableHighlight>
          </Component>,
          "style": Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                Hello World!
              </Text>,
              <TouchableHighlight
                activeOpacity={0.85}
                delayPressOut={100}
                onPress={[MockFunction]}
                underlayColor="black"
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  Hide Modal
                </Text>
              </TouchableHighlight>,
            ],
            "style": Object {
              "backgroundColor": "#aacadd",
              "height": "50%",
              "width": "90%",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Hello World!",
                "ellipsizeMode": "tail",
              },
              "ref": null,
              "rendered": "Hello World!",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "activeOpacity": 0.85,
                "children": <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  Hide Modal
                </Text>,
                "delayPressOut": 100,
                "onPress": [MockFunction],
                "underlayColor": "black",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "Hide Modal",
                  "ellipsizeMode": "tail",
                },
                "ref": null,
                "rendered": "Hide Modal",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "Modal",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
